File: Algorithms/Search Sort/Algorithms/selectionSort/SelectionSort.java
class SelectionSort {
  public static void main(String args[]) {
    int[] arr = { 1, 7, 9, 4, 2, 4 };
    int len = arr.length;
    selectionSort(arr, len);
    for (int i : arr) {
      System.out.println(i + " ");
    }
  }

  public static void selectionSort(int[] arr, int len) {
    int i = 0, j, lowIdx = 0, temp = 0;
    for (i = 0; i < len - 1; i++) {
      lowIdx = i;
      for (j = i; j < len; j++) {
        if (arr[j] < arr[lowIdx]) {
          lowIdx = j;
        }
      }
      temp = arr[lowIdx];
      arr[lowIdx] = arr[i];
      arr[i] = temp;
    }
  }
}

File: Algorithms/Search Sort/Algorithms/selectionSort/selectionSort.ts

const selectionSort = (arr: number[], size: number) => {

  const tuple: [number, number, number, number] = [0, 0, 0, 0]
  let [i, j, menorIndex, temp] = tuple;

  for (i = 0; i < size - 1; i++){
    menorIndex = i;

    for (j = i; j < size; j++){
      if (arr[j] < arr[menorIndex]) {
        menorIndex = j;
      }
    }
    
    temp = arr[menorIndex];
    arr[menorIndex] = arr[i];
    arr[i] = temp;
  }
  print(arr);
}

function print(arr: number[]) {
  arr.forEach(el => {
    console.log(el)
  })
}

(() => {
  const arr: number[] = [1, 4, 6, 8, 3, 1, 4, 56];
  const size: number = arr.length;
  selectionSort(arr, size);
})()
